src/compas_timber/fabrication/btlx.py

```diff
@@ -52,112 +48,173 @@ class BTLx(object):
-            "ComputerName": "{}".format(os.getenv("computername")),
-            "UserName": "{}".format(os.getenv("USERNAME")),
-            "FileName": "",
-            "Date": "{}".format(date.today()),
-            "Time": "{}".format(datetime.now().strftime("%H:%M:%S")),
-            "Comment": "",
-        }
+    def __init__(self, company_name=None, file_name=None, comment=None):
+        self.company_name = company_name
+        self.file_name = file_name
+        self.comment = comment
 
-    def btlx_string(self):
-        """Returns a pretty XML string for visualization in GH, Terminal, etc."""
-        self.ET_element = ET.Element("BTLx", BTLx.FILE_ATTRIBUTES)
-        self.ET_element.append(self.file_history)
-        self.project_element = ET.SubElement(self.ET_element, "Project", Name="testProject")
-        self.parts_element = ET.SubElement(self.project_element, "Parts")
-
-        for part in self.parts.values():
-            self.parts_element.append(part.et_element)
-        return MD.parseString(ET.tostring(self.ET_element)).toprettyxml(indent="   ")
-
-    def process_model(self):
-        """Processes the model and generates BTLx parts."""
-        for index, beam in enumerate(self.model.beams):
-            self.parts[str(beam.guid)] = BTLxPart(beam, order_num=index)
-
-        for joint in self.joints:
-            factory_type = self.REGISTERED_JOINTS.get(str(type(joint)))
-            if factory_type is None:
-                continue  # that way we can just unregister factories that are replaced by the new features
-            factory_type.apply_processings(joint, self.parts)
-
-        # TODO: to slowly integrate the new system, iterate here once more, this time on beams and their features
-        # add processings from features that are part of the new system AKA have the attribute `PROCESS_NAME`
-        # joints that are already part of the new system should be skipped above (e.g. L-Miter)
-        for beam in self.model.beams:
-            features = list(filter(lambda feature: hasattr(feature, "PROCESS_NAME"), beam.features))
-            beam_part = self.parts[str(beam.guid)]
-            self._apply_process_features(beam_part, features)
-
-    def _apply_process_features(self, beam_part, features):
-        for feature in features:
-            # Create BTLXProcess instance from feature
-            header_dict, process_dict = self._split_params_dict(feature)
-            process = BTLxProcess(feature.PROCESS_NAME, header_dict, process_dict)
-            # append to beam_part.processings
-            beam_part.processings.append(process)
-
-    def _split_params_dict(self, feature):
-        whole_dict = feature.params_dict
-        header_keys = [
-            "Name",
-            "Process",
-            "Priority",
-            "ProcessID",
-            "ReferencePlaneID",
-        ]
+    def write(self, model, file_path):
+        """Writes the BTLx file to the given file path.
```

Comment:
add `See Also` to `model_to_xml`

Reply:
e.g. https://github.com/compas-dev/compas/blob/7bbe6acc032e8bc1fa8f80ddcb66d41fb8421f3e/src/compas/geometry/curves/bezier.py#L142